src/components/ha-md-dialog.ts

```diff
@@ -0,0 +1,96 @@
+import { MdDialog } from "@material/web/dialog/dialog";
+import { css } from "lit";
+import { customElement, property } from "lit/decorators";
+
+/**
+ * Based on the home assistant design: https://design.home-assistant.io/#components/ha-dialogs
+ *
+ */
+@customElement("ha-md-dialog")
+export class HaMdDialog extends MdDialog {
+  /**
+   * When true the dialog will not close when the user presses the esc key or press out of the dialog.
+   */
+  @property({ attribute: "disable-cancel-action", type: Boolean })
+  public disableCancelAction = false;
+
+  constructor() {
+    super();
+    this.open = true;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener("cancel", this._handleCancel);
+  }
+
+  _handleCancel(closeEvent: Event) {
+    if (this.disableCancelAction) {
+      closeEvent.preventDefault();
+      const dialogElement = this.shadowRoot?.querySelector("dialog");
+      dialogElement?.animate(
```

Comment:
I wonder if we should use the `animate` API, since that will not be supported by older browsers. (if we do, we should check for support first)
This seems like an easy enough animation to just do in css?

Reply:
![image](https://github.com/user-attachments/assets/84543c11-577d-4f2c-8324-0ab4d4277f6f)
94.3% supports it.

The question is what happens when it fails, When it just ignores the animation this is fine.

Reply:
Agree, test on iOS 12 :-)